The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12314794316/job/34371655885) [(plain)](https://github.com/rust-lang/rust/commit/a2672ef9565a6241e9c48bf55b6108f1ed7af1d8/checks/34371655885/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.725 Building wheels for collected packages: reuse
#13 2.726   Building wheel for reuse (pyproject.toml): started
#13 2.970   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.971   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 2.972   Stored in directory: /tmp/pip-ephem-wheel-cache-t4d8xyij/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.974 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.373 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.374 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.913 Collecting virtualenv
#13 3.913 Collecting virtualenv
#13 3.984   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.093      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 40.3 MB/s eta 0:00:00
#13 4.133 Collecting distlib<1,>=0.3.7
#13 4.148   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.157      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 69.4 MB/s eta 0:00:00
#13 4.193 Collecting platformdirs<5,>=3.9.1
#13 4.207   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.244 Collecting filelock<4,>=3.12.2
#13 4.259   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.341 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.522 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      333760 kB
DirectMap2M:     9103360 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
fmt check
Diff in /checkout/library/core/src/error.rs:2:
 #![stable(feature = "error_in_core", since = "1.81.0")]
 
 use crate::any::TypeId;
-use crate::fmt::{Debug, Display, Formatter, self};
+use crate::fmt::{self, Debug, Display, Formatter};
 /// `Error` is a trait representing the basic expectations for error values,
 /// `Error` is a trait representing the basic expectations for error values,
 /// i.e., values of type `E` in [`Result<T, E>`].
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/future/join.rs" "/checkout/library/core/src/future/ready.rs" "/checkout/library/core/src/future/mod.rs" "/checkout/library/core/src/ascii.rs" "/checkout/library/core/src/cmp.rs" "/checkout/library/core/src/ascii/ascii_char.rs" "/checkout/library/core/src/option.rs" "/checkout/library/core/src/asserting.rs" "/checkout/library/core/src/any.rs" "/checkout/library/core/src/pat.rs" "/checkout/library/core/src/clone.rs" "/checkout/library/core/src/pin.rs" "/checkout/library/core/src/bool.rs" "/checkout/library/core/src/fmt/builders.rs" "/checkout/library/core/src/fmt/nofloat.rs" "/checkout/library/core/src/fmt/float.rs" "/checkout/library/core/src/fmt/num.rs" "/checkout/library/core/src/fmt/rt.rs" "/checkout/library/core/src/fmt/mod.rs" "/checkout/library/core/src/error.rs" "/checkout/library/core/src/net/parser.rs" "/checkout/library/core/src/net/ip_addr.rs" "/checkout/library/core/src/net/display_buffer.rs" "/checkout/library/core/src/net/socket_addr.rs" "/checkout/library/core/src/net/mod.rs" "/checkout/library/core/src/ptr/mut_ptr.rs" "/checkout/library/core/src/ptr/const_ptr.rs" "/checkout/library/core/src/ptr/non_null.rs" "/checkout/library/core/src/ptr/alignment.rs" "/checkout/library/core/src/ptr/metadata.rs" "/checkout/library/core/src/ptr/unique.rs" "/checkout/library/core/src/ptr/mod.rs" "/checkout/src/ci/docker/host-x86_64/test-various/uefi_qemu_test/src/main.rs" "/checkout/library/core/src/num/f16.rs" "/checkout/library/core/src/num/nonzero.rs" "/checkout/library/core/src/num/f128.rs" "/checkout/library/core/src/num/int_macros.rs" "/checkout/library/core/src/num/saturating.rs" "/checkout/library/core/src/num/shells/int_macros.rs" "/checkout/library/core/src/num/shells/i128.rs" "/checkout/library/core/src/num/shells/u64.rs" "/checkout/library/core/src/num/shells/u8.rs" "/checkout/library/core/src/num/shells/i16.rs" "/checkout/library/core/src/num/shells/isize.rs" "/checkout/library/core/src/num/shells/u128.rs" "/checkout/library/core/src/num/shells/i8.rs" "/checkout/library/core/src/num/shells/u32.rs" "/checkout/library/core/src/num/shells/usize.rs" "/checkout/library/core/src/num/shells/i32.rs" "/checkout/library/core/src/num/shells/u16.rs" "/checkout/library/core/src/num/shells/i64.rs" "/checkout/library/core/src/num/fmt.rs" "/checkout/library/core/src/num/error.rs" "/checkout/library/core/src/num/overflow_panic.rs" "/checkout/library/core/src/num/flt2dec/decoder.rs" "/checkout/library/core/src/num/flt2dec/estimator.rs" "/checkout/library/core/src/num/flt2dec/strategy/grisu.rs" "/checkout/library/core/src/num/flt2dec/strategy/dragon.rs" "/checkout/library/core/src/num/flt2dec/mod.rs" "/checkout/library/core/src/num/bignum.rs" "/checkout/library/core/src/num/uint_macros.rs" "/checkout/library/core/src/num/f64.rs" "/checkout/library/core/src/num/wrapping.rs" "/checkout/library/core/src/future/future.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri Dec 13 11:39:34 UTC 2024
  network time: Fri, 13 Dec 2024 11:39:35 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/error.rs
+++ b/library/core/src/error.rs
@@ -2,7 +2,7 @@
 #![stable(feature = "error_in_core", since = "1.81.0")]
 
 use crate::any::TypeId;
-use crate::fmt::{Debug, Display, Formatter, Result};
+use crate::fmt::{self, Debug, Display, Formatter};
 
 /// `Error` is a trait representing the basic expectations for error values,
 /// i.e., values of type `E` in [`Result<T, E>`].
@@ -857,7 +857,7 @@ impl<'a> Request<'a> {
 
 #[unstable(feature = "error_generic_member_access", issue = "99301")]
 impl<'a> Debug for Request<'a> {
-    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         f.debug_struct("Request").finish_non_exhaustive()
     }
 }

